enterprise/internal/insights/background/retention/worker.go

```diff
@@ -7,36 +7,72 @@ import (
 	"github.com/keegancsmith/sqlf"
 	"github.com/sourcegraph/log"
 
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/insights/store"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
 	"github.com/sourcegraph/sourcegraph/internal/observation"
 	"github.com/sourcegraph/sourcegraph/internal/workerutil"
 	"github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker"
 	dbworkerstore "github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker/store"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
 var _ workerutil.Handler[*DataRetentionJob] = &dataRetentionHandler{}
 
 type dataRetentionHandler struct {
 	baseWorkerStore dbworkerstore.Store[*DataRetentionJob]
+	insightsStore   *store.Store
 }
 
-func (h *dataRetentionHandler) Handle(ctx context.Context, logger log.Logger, record *DataRetentionJob) error {
-	logger.Debug("data retention handler called", log.Int("seriesID", record.SeriesID))
+func (h *dataRetentionHandler) Handle(ctx context.Context, logger log.Logger, record *DataRetentionJob) (err error) {
+	// Default should match what is shown in the schema not to be confusing
+	maximumSampleSize := 90
+	if configured := conf.Get().InsightsMaximumSampleSize; configured != 0 {
```

Comment:
```suggestion
	if configured := conf.Get().InsightsMaximumSampleSize; configured >= 0 {
```
minor: If for some reason someone added a negative value it will break the sql OFFSET.